geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/InternalGfshCommand.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.internal.cli.commands;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.springframework.shell.core.CommandMarker;
+
+import org.apache.geode.management.ManagementService;
+import org.apache.geode.management.cli.GfshCommand;
+import org.apache.geode.management.cli.Result;
+import org.apache.geode.management.internal.cli.functions.CliFunctionResult;
+import org.apache.geode.management.internal.cli.shell.Gfsh;
+import org.apache.geode.management.internal.configuration.domain.XmlEntity;
+
+/**
+ * Encapsulates common functionality for implementing command classes for the Geode shell (gfsh).
+ * this provides wrapper around the static methods in CliUtils for easy mock of the commands
+ *
+ * this class should not have much implementation of its own other then those tested in
+ * GfshCommandJUnitTest
+ */
+@SuppressWarnings("unused")
+public abstract class InternalGfshCommand extends GfshCommand implements CommandMarker {
```

Comment:
Does not need to implement CommandMarker if the public GfshCommand definition is changed to implement CommandMarker

Reply:
Done

Reply:
Parsing commands fail if `InternalGfshCommand` does not implement CommandMarker.